stablehlo/dialect/TypeInference.cpp

```diff
@@ -502,6 +503,16 @@ LogicalResult verifyReducerShape(
 // Shape functions for ops.
 //===----------------------------------------------------------------------===//
 
+LogicalResult inferAfterAllOp(MLIRContext* context, Optional<Location> location,
+                              ValueRange inputs,
+                              SmallVectorImpl<Type>& inferredReturnTypes) {
+  if (inputs.size() == 0)
+    inferredReturnTypes.emplace_back(stablehlo::TokenType::get(context));
+  else
+    inferredReturnTypes.emplace_back(inputs[0].getType());
```

Comment:
`emplace_back` is useful for vectors of `ShapedTypeComponent` because it constructs a vector element from the passed arguments, so we don't have to write `ShapedTypeComponent(arg0, arg1, ...)`. 

Since we have a vector of `Type`, we should use `push_back` here, because we don't need to construct a vector element here - we just want to insert it. `emplace_back` still works because `Type` has a copy constructor, but it's redundant.